packages/ui/src/components/notifications/NotificationCenter.test.tsx

```diff
@@ -0,0 +1,169 @@
+import { act, fireEvent, render, waitFor } from '@testing-library/react';
+import { describe, expect, it, vi } from 'vitest';
+
+import NotificationCenter, { notify } from './NotificationCenter';
+import { Notification } from './types';
+
+type SnackbarProps = Notification & { onClose: (id: string) => void };
+
+function randomString() {
+    return Math.random().toString(36).substring(2, 15);
+}
+const mockRandomUUID = vi.fn(() => {
+    return `${randomString()}-${randomString()}-${randomString()}-${randomString()}-${randomString()}`;
+});
+global.crypto = {
+    subtle: global.crypto.subtle,
+    getRandomValues: vi.fn(),
+    randomUUID: mockRandomUUID as () => `${string}-${string}-${string}-${string}-${string}`,
+};
+const NOTIFICATION_EVENT = 'emit-core-notification';
+
+describe('NotificationCenter', () => {
+    const mockNotification = (message: string) => {
```

Comment:
let's rename it to `generateMockNotification` since it generates a random one every time